src/aliceVision/lensCorrectionProfile/lcp.hpp

```diff
@@ -0,0 +1,733 @@
+    inline void closeAlternateLensIDs() { _inAlternateLensIDs = false; }
+
+    inline bool isAlternateLensNamesOpened() { return _inAlternateLensNames; }
+    inline void openAlternateLensNames() { _inAlternateLensNames = true; }
+    inline void closeAlternateLensNames() { _inAlternateLensNames = false; }
+
+    inline bool isWaitPerspModeldescription() { return _waitPerspModeldescription; }
+    inline void setWaitPerspModeldescription() { _waitPerspModeldescription = true; }
+    inline void unsetWaitPerspModeldescription() { _waitPerspModeldescription = false; }
+
+    inline bool isGetText() { return _getText; }
+    inline void setGetText() { _getText = true; }
+    inline void unsetGetText() { _getText = false; }
+
+    inline int getModelCount() { return _modelCount; }
+    inline void increaseModelCount() { _modelCount++; }
+
+    inline void storeCurrParams() { v_lensParams.push_back(currLensParam); }
+
+    LensParam currLensParam;
+
+    bool search(settingsInfo& settings, LCPCorrectionMode mode, int& iLow, int& iHigh, float& weightLow);
+    void combine(size_t iLow, size_t iHigh, float weightLow, LCPCorrectionMode mode, LensParam& pOut);
+
+    // Loading states
+    bool _isSeqOpened = false;
+    bool _isCommonOK = false;
+    bool _isCamDataOK = false;
+    bool _inAlternateLensIDs = false;
+    bool _inAlternateLensNames = false;
+    bool _waitPerspModeldescription = false;
+    bool _getText = false;
+    int _modelCount = 0;
+
+    LCPReadingState _currReadingState = LCPReadingState::WaitSequence;
+    std::string _currText = "";
+
+    // Set of models contained in the LCP file
+    std::vector<LensParam> v_lensParams;
+
+    // Camera and Lens information, common for all models
+    std::string Author = "";
+    std::string Make = "";
+    std::string Model = "";
+    std::string UniqueCameraModel = "";
+    bool CameraRawProfile;
+    std::vector<int> LensID;
+    std::vector<std::string> Lens;
+    std::string LensInfo = "";
+    std::string CameraPrettyName = "";
+    std::string LensPrettyName = "";
+    std::string ProfileName = "";
+    float SensorFormatFactor = 1.f;
+    int ImageWidth = 0;
+    int ImageLength = 0;
+    float XResolution = 0.f;
+    float YResolution = 0.f;
+
+    void setCommonSettings(const std::string& name)
+    {
+        if (name == "stCamera:Author")
+            Author = _currText;
+        else if (name == "stCamera:ProfileName")
+            ProfileName = _currText;
+        else if (name == "stCamera:Make")
+            Make = _currText;
+        else if (name == "stCamera:Model")
+            Model = _currText;
+        else if (name == "stCamera:Lens")
+            Lens.push_back(_currText);
+        else if (name == "stCamera:LensID")
+            LensID.push_back(std::atoi(_currText.c_str()));
+        else if (name == "stCamera:LensInfo")
+            LensInfo = _currText;
+        else if (name == "stCamera:ImageWidth")
+            ImageWidth = atoi(_currText.c_str());
```

Comment:
`atoi` does not detect errors, maybe use `std::stoi` instead (same for `atof`, etc.)

Reply:
Done